### PR TITLE
Move INVARIANT() definition to hse_util/assert.h

### PR DESCRIPTION
**Pull request by tristan957**
Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

**Review by alexttx — APPROVED** [review]

File: lib/kvdb/ikvdb.c [hunk 1]
```diff
@@ -1969,13 +1970,14 @@ ikvdb_kvs_open(
     params->read_only = self->ikdb_rp.read_only; /* inherit from kvdb */
 
     for (i = HSE_MCLASS_BASE; i < HSE_MCLASS_COUNT; i++) {
-        if (strstr(params->mclass_policy, mpool_mclass_to_string[i])) {
+        const char *name = hse_mclass_name_get(i);
+        if (strstr(params->mclass_policy, name)) {
```
Comment (beckerg) [inline lib/kvdb/ikvdb.c:1974]: Leave a blank line after decls..

File: lib/util/include/hse_util/condvar.h [hunk 2]
```diff
@@ -16,6 +16,8 @@
 #ifndef HSE_PLATFORM_CONDVAR_H
 #define HSE_PLATFORM_CONDVAR_H
 
+#include <stdlib.h>
+
```
Comment (beckerg) [inline lib/util/include/hse_util/condvar.h:20]: Why is stdlib.h suddenly required by condvar.h?

File: lib/util/include/hse_util/mutex.h [hunk 3]
```diff
@@ -64,6 +64,10 @@
  * descheduling.
  */
 
+#include <errno.h>
+#include <pthread.h>
+#include <stdlib.h>
+
```
Comment (beckerg) [inline lib/util/include/hse_util/mutex.h:70]: Not clear what in mutex.h needs errno.h nor stdlib.h.  But it always should have had pthread.!

**Review by beckerg — APPROVED** [review]

File: lib/util/include/hse_util/mutex.h (hunk 3 above)
Comment (tristan957) [reply lib/util/include/hse_util/mutex.h:70]: There is an EBUSY reference and abort() reference.

File: lib/util/include/hse_util/condvar.h (hunk 2 above)
Comment (tristan957) [reply lib/util/include/hse_util/condvar.h:20]: abort()

Comment (beckerg) [reply lib/util/include/hse_util/condvar.h:20]: Needs pthread.h too!

**Pull request merged**